CMakeLists.txt

```diff
@@ -51,11 +60,12 @@ else()
 endif()
 list(APPEND package_deps thrift)
 
+
 hunter_add_package(opentracing-cpp)
 # Not `${hunter_config}` because OpenTracing provides its own
 # OpenTracingConfig.cmake file
 find_package(OpenTracing CONFIG REQUIRED)
-list(APPEND LIBS OpenTracing::opentracing-static)
+list(APPEND LIBS OpenTracing::opentracing)
```

Comment:
Could OpenTracing static linking be provided as an option? It's necessary to make the plugin work with projects that are themselves statically linked or in environments without the c++ standard library installed.

See https://stackoverflow.com/q/47841812/4447365 and https://github.com/envoyproxy/envoy/pull/2252

Reply:
The problem I faced with the static link to Open Tracing was that the OT Global Tracer was defined many times. For example, it was defined in the unit test executable and in the Jaeger library.

I can change the code to keep the link as before on Linux and the dynamic link on Windows.

What do you think?

Reply:
I'd be fine with that.

I think there are scenarios where you'd want to link dynamically on linux also, but up to @isaachier how he'd like to support them.

Reply:
Hey I must have totally missed this PR. Will look ASAP.

Reply:
Ya ideally use `MSVC` or the like in `CMakeLists.txt` to make this decision.

Reply:
changed. With MSVC, a shared library is used. Otherwise, it'll be a static one.